app/forms/questionnaire_form.py

```diff
@@ -25,29 +31,82 @@ def __init__(self, schema, block_json, formdata=None, **kwargs):
 
     def validate(self):
         """
-        Validate this form as usual and check for any form-level date-range validation errors
+        Validate this form as usual and check for any form-level date-range/calculated validation errors
         :return:
         """
-        valid_form = True
         valid_fields = FlaskForm.validate(self)
 
         for question_id, period_from_id, period_to_id in self.date_ranges:
-            if period_from_id not in self.errors and period_to_id not in self.errors:
-                period_from = getattr(self, period_from_id)
-                period_to = getattr(self, period_to_id)
-                validator = DateRangeCheck()
+            period_ids = [period_from_id, period_to_id]
+            if not (
+                    self.answers_all_valid(period_ids) and
+                    self._validate_date_range_question(question_id, period_from_id, period_to_id)):
+                return False
+
+        for question, target_answer in self.calculated_question:
+            target_total = self.answer_store.filter(answer_ids=[question['calculated'].get('answer_id')]).values()[0]
+            currency = target_answer.get('currency')
+            if not (
+                    self.answers_all_valid(question['calculated']['answers_to_calculate']) and
+                    self._validate_calculated_question(question['id'], question['calculated'], target_answer['label'],
+                                                       target_total, currency)):
+                return False
+
+        return True and valid_fields
```

Comment:
`return True and valid_fields` is the same as `return valid_fields`. It might be worth putting back in the concept of `valid_form` and set those above rather than returning early. You'd then `return valid_fields and valid_form`